include/podio/utilities/TypeHelpers.h

```diff
@@ -244,8 +247,63 @@ class CollectionBase;
 
 /// Concept for checking whether a passed type T inherits from podio::CollectionBase
 template <typename T>
-concept CollectionType = std::is_base_of_v<CollectionBase, T>;
-
+concept CollectionType = requires(T t, const T ct) {
+  // general concepts
+  requires !std::is_abstract_v<T>;
+  requires std::derived_from<T, CollectionBase>;
+  requires std::default_initializable<T>;
+  requires std::destructible<T>;
+  requires std::movable<T>;
+  requires !std::copyable<T>;
+  requires std::ranges::random_access_range<T>;
+  // typeName's
+  { T::typeName } -> std::convertible_to<std::string_view>;
+  { std::bool_constant<(T::typeName, true)>() } -> std::same_as<std::true_type>; // ~is annotated with constexpr
+  { T::valueTypeName } -> std::convertible_to<std::string_view>;
+  { std::bool_constant<(T::valueTypeName, true)>() } -> std::same_as<std::true_type>; // ~is annotated with constexpr
+  { T::dataTypeName } -> std::convertible_to<std::string_view>;
+  { std::bool_constant<(T::dataTypeName, true)>() } -> std::same_as<std::true_type>; // ~is annotated with constexpr
+  // typedefs
+  typename T::value_type;
+  typename T::mutable_type;
+  requires std::convertible_to<typename T::mutable_type, typename T::value_type>;
```

Comment:
I think for this one and the ones below the requirement of the typename to exist doesn't need to be there before since it won't compile when it reaches this `requires`. I guess it's there for explicitness.

Reply:
I suppose in most of the cases this shouldn't really matter as the majority of the collections are generated with the right interface in any case. I do like the more explicit  message a bit better though, even if the information content / readability is probably the same at the point where you actually have to look at these errors ;)